### PR TITLE
[ingress-nginx] add nginx-ingress v1.6.4

### PR DESCRIPTION
**Pull request by fl64**
## Description

Added nginx ingress controller `v1.6.4`

Fragment list of changes for nginx, nginx-controller to pay attention to when using this controller.

[nginx-controller](https://raw.githubusercontent.com/kubernetes/ingress-nginx/controller-v1.6.4/Changelog.md):
* Upgrade NGINX to `1.21.6`
* Ingress-nginx now is using `Endpointslices` instead of `Endpoints`. 
* Update to Prometheus metric names, more information [available here]( https://github.com/kubernetes/ingress-nginx/pull/8728
)
* Deprecated Kubernetes versions `1.20`-`1.21`, Added support for, `1.25`, currently supported versions `v1.22`, `v1.23`, `v1.24`, `v1.25`  
* This release removes the `root` and `alias` directives in NGINX, this can avoid some potential security attacks.
* This release also brings a special new feature of deep inspection into objects. The inspection is a walk through of all the spec, checking for possible attempts to escape configs. Currently such an inspection only occurs for `networking.Ingress`.

[nginx](http://nginx.org/en/CHANGES-1.22):
* Feature: the `"proxy_half_close"` directive in the stream module.
* Feature: the `"ssl_alpn"` directive in the stream module.
* Feature: the `"mp4_start_key_frame"` directive in the `ngx_http_mp4_module`.
* Bugfix: requests might hang when using HTTP/2 and the `"aio_write"` directive.
* Bugfix: the security level, which is available in OpenSSL 1.1.0 or newer, did not affect loading of the server certificates when set        with "@SECLEVEL=N" in the `"ssl_ciphers" `directive.
* Security: 1-byte memory overwrite might occur during DNS server response processing if the `"resolver" `directive was used, allowing an attacker who is able to forge UDP packets from the DNS server to cause worker process crash or, potentially, arbitrary code execution (CVE-2021-23017).
* Feature: variables support in the` "proxy_ssl_certificate"`, `"proxy_ssl_certificate_key"` `"grpc_ssl_certificate"`, `"grpc_ssl_certificate_key"`, `"uwsgi_ssl_certificate"`, and `"uwsgi_ssl_certificate_key"` directives.
* Feature: the `"max_errors"` directive in the mail proxy module.
* Feature: the `"fastopen"` parameter of the `"listen"` directive in the Stream module.

## What is the expected result?
To use controller, set `controllerVersion: "1.6"` in `deckhouse.io/v1` `IngressNginxController` resource

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: ingress-nginx
type: feature
summary: | 
Added nginx ingress controller v1.6.4
nginx-controller:
* Upgrade NGINX to 1.21.6
* Ingress-nginx now is using Endpointslices instead of Endpoints. 
* Update to Prometheus metric names, more information [available here]( https://github.com/kubernetes/ingress-nginx/pull/8728
)
* Deprecated Kubernetes versions 1.20-1.21, Added support for, 1.25, currently supported versions v1.22, v1.23, v1.24, v1.25  
* This release removes the `root` and `alias` directives in NGINX, this can avoid some potential security attacks.
* This release also brings a special new feature of deep inspection into objects. The inspection is a walk through of all the spec, checking for possible attempts to escape configs. Currently such an inspection only occurs for `networking.Ingress`.
nginx:
* Feature: the "proxy_half_close" directive in the stream module.
* Feature: the "ssl_alpn" directive in the stream module.
* Feature: the "mp4_start_key_frame" directive in the ngx_http_mp4_module.
* Bugfix: requests might hang when using HTTP/2 and the "aio_write" directive.
* Bugfix: the security level, which is available in OpenSSL 1.1.0 or newer, did not affect loading of the server certificates when set        with "@SECLEVEL=N" in the "ssl_ciphers" directive.
* Security: 1-byte memory overwrite might occur during DNS server response processing if the "resolver" directive was used, allowing an attacker who is able to forge UDP packets from the DNS server to cause worker process crash or, potentially, arbitrary code execution (CVE-2021-23017).
* Feature: variables support in the "proxy_ssl_certificate", "proxy_ssl_certificate_key" "grpc_ssl_certificate", "grpc_ssl_certificate_key", "uwsgi_ssl_certificate", and "uwsgi_ssl_certificate_key" directives.
* Feature: the "max_errors" directive in the mail proxy module.
* Feature: the "fastopen" parameter of the "listen" directive in the Stream module.
impact_level: low
```

### CODE REVIEW

File: modules/402-ingress-nginx/templates/controller/controller.yaml [hunk 1]
```diff
@@ -293,6 +293,8 @@ spec:
           name: uwsgi-temp-path
         - mountPath: /etc/nginx/ssl/
           name: secret-nginx-auth-tls
+        - mountPath: /tmp/nginx/
+          name: tmp-nginx
```
Comment (yalosev) [inline modules/402-ingress-nginx/templates/controller/controller.yaml:297]: Is it useful? Seems like you left something for debug here

**Review by yalosev — COMMENTED** [review]
A few questions

File: modules/402-ingress-nginx/templates/controller/configmap.yaml [hunk 2]
```diff
@@ -84,6 +84,7 @@ data:
                 AES256-SHA:AES128-SHA:DHE-RSA-AES256-SHA:DHE-RSA-AES128-SHA"
   {{- end }}
   server-tokens: "false"
+  global-auth-always-set-cookie: "true"
```
[the remark this answers is not in the document]
Comment (fl64) [reply modules/402-ingress-nginx/templates/controller/configmap.yaml:87]: We got rid of this auth-cookie-always.patch and this parameter will enable the logic that the patch previously performed

File: modules/402-ingress-nginx/templates/controller/controller.yaml (hunk 1 above)
Comment (fl64) [reply modules/402-ingress-nginx/templates/controller/controller.yaml:297]: At least a mount point is needed to create a prometheus-nginx.socket

File: modules/402-ingress-nginx/openapi/values.yaml [hunk 3]
```diff
@@ -74,7 +74,7 @@ properties:
                   x-examples: ["HostWithFailover"]
                 controllerVersion:
                   type: string
-                  x-examples: ["1.1"]
+                  x-examples: ["1.6"]
```
[the remark this answers is not in the document]
Comment (fl64) [reply modules/402-ingress-nginx/openapi/values.yaml]: fixed

**Review by yalosev — APPROVED** [review]

**Review by nabokihms — APPROVED** [review]

**Pull request merged**